presto-main/src/main/java/com/facebook/presto/Session.java

```diff
@@ -138,6 +143,7 @@ public Session(
         checkArgument(!transactionId.isPresent() || unprocessedCatalogProperties.isEmpty(), "Catalog session properties cannot be set if there is an open transaction");
 
         checkArgument(catalog.isPresent() || !schema.isPresent(), "schema is set but catalog is not");
+        this.sessionLogger = this.queryLoggingSize > 0 ? new SizeLimitedSessionLogger(this.queryId, this.queryLoggingSize) : SessionLogger.NOOP;
```

Comment:
If we changed the name to `NOOP_SESSION_LOGGER`, let's also do static import here.